Browserify and CommonMark

KyGost opened the pull request:
Use Browserify to use packaged content instead of inline minified javascript
Use CommonMark for common markdown standard

KyGost committed:
Browserify and CommonMark
Use Browserify to use packaged content instead of inline minified javascript
Use CommonMark for common markdown standard

KyGost committed:
Remove semicolons for standard

RangerMauve commented on script.js:
```diff
@@ -1,56 +1,57 @@
+
+        let walker = parsed.walker()
+        while(walker.current.type !== 'heading') walker.next()
+        let title = (walker.current.type === 'heading') ? walker.current.firstChild.literal : 'Markdown Document' // Find relevant page content here?
```
Does this work? Maybe the default title could be the file name?

KyGost replied:
This works.

Default being file name sounds good.

RangerMauve reviewed (changes requested):
Cool, this looks good to me!

Could you maybe add a `build` command to the package.json's `scripts` section? Or I guess now we need a package.json to track the dependencies. 😅 

Also, mind talking a bit about what the advantage of using commonmark is over the previous markdown renderer?

KyGost commented:
What's the build command? I'll google.

It is standardised, that's the main advantage.

RangerMauve commented:
You'll want to use `npm init` to initialize a package.json, use `npm i --save commonmark` to add it as a dependency, use `npm i --save-dev browserify` to add it as a dev dependency, and then add a "scripts" section to the package.json which has an item called `build` set to something like `browserify ./script > bundle.js` so we can use `npm build` to build the bundle.

KyGost commented:
Oh awesome, will do.

KyGost committed:
Build script
Add build script

KyGost commented:
Updated

RangerMauve commented on script.js:
```diff
@@ -1,56 +1,57 @@
+        let rendered = renderer.render(parsed)
+
+        let walker = parsed.walker()
+        while(walker.current.type !== 'heading') walker.next()
```
Should account for there being no headings in the document. Is there a `walker.hasNext`?

KyGost replied:
Ah, you make a good point haha.
I'll refine the code.

KyGost replied:
Fixed

RangerMauve reviewed (changes requested):
Minor comment about a potential infinite loop in the walker, but this is looking good!

KyGost commented:
In trying to get this to work, it wasn't working... I was baffled by the cause... Until I checked the source.
![image](https://user-images.githubusercontent.com/12588894/96818203-b7f14d00-146c-11eb-91bf-8be74026eb30.png)
With the current method for finding out if the content looks like a plaintext page, it fails if another extension has gotten to it first.

Any ideas on how to avoid this? Any way to set priority? Or perhaps we are less strict with how we decide if something is markdown?

Perhaps we could, rather than looking for a document that contains **only** a `<pre>`, we look for a tag that contains, as a child of body, the `<pre>` tag, with the funky styles that it seems to get (`style="word-wrap: break-word; white-space: pre-wrap;"`).

KyGost committed:
Protection
Protection from no heading (walker.current becomes null)
Protection from other extensions which inject HTML
Fixed package.json

KyGost commented:
Solved and tested

RangerMauve reviewed (approved):
Nice, this looks perfect. Thank you!
